packages/backend/src/ee/services/ai/utils/validators.ts

```diff
@@ -211,16 +219,29 @@ function validateFilterRule(filterRule: FilterRule, field: CompiledField) {
     }
 
     try {
-        renderFilterRuleSqlFromField(
-            filterRule,
-            field,
-            // ! The following args are used to actually render the SQL, we don't care about the ouput, just that it doesn't throw
-            '"',
-            "'",
-            (string: string) => string.replaceAll('\\', '\\\\'),
-            WeekDay.SUNDAY,
-            SupportedDbtAdapter.BIGQUERY,
-        );
+        if (isAdditionalMetric(field)) {
+            renderFilterRuleSql(
+                filterRule,
+                field.type,
+                field.sql,
+                // ! The following args are used to actually render the SQL, we don't care about the ouput, just that it doesn't throw
+                '"',
+                (string: string) => string.replaceAll('\\', '\\\\'),
+                WeekDay.SUNDAY,
+                SupportedDbtAdapter.BIGQUERY,
+            );
```

Comment:
`renderFilterRuleSqlFromField` only accepts compiled fields, so I reference a util that is defined inside